gnovm/pkg/gnolang/values.go

```diff
@@ -1969,9 +1970,8 @@ func (tv *TypedValue) GetPointerAtIndex(rlm *Realm, alloc *Allocator, store Stor
 			}
 		}
 		// attach mapkey object
-		if !exist {
-			rlm.DidUpdate(mv, nil, iv.GetFirstObject(store))
-		}
+		rlm.DidUpdate(mv, oldObject, ivk.GetFirstObject(store))
+
```

Comment:
trying to understand why it's necessary to do this part... investigating.

Reply:
See: https://go.dev/play/p/iNPDR4FQlRv

The last store into a map index sets the real "key" in the map, so it needs to be updated each time to match go's behaviour.